data/shared/citizen/scripting/v8/main.js

```diff
@@ -463,30 +463,31 @@ const EXT_LOCALFUNCREF = 11;
 				global.source = parseInt(source.substr(13));
 			}
 
-			const payload = unpack(payloadSerialized) || [];
 			const listeners = emitter.listeners(name);
-
-			if (listeners.length === 0 || !Array.isArray(payload)) {
-				global.source = null;
-				return;
-			}
-
-			// Running normal event listeners
-			for (const listener of listeners) {
-				try {
-					const retval = listener.apply(null, payload);
-
-					if (retval instanceof Promise) {
-						(async () => {
-							try {
-								await retval;
-							} catch (e) {
-								console.error('Unhandled promise failure:', e);
-							}
-						})();
+			if (listeners.length > 0) {
```

Comment:
Prefer to see this being applied as an early out, rather than more scoping. Especially to reduce the amount of lines being changed.